### PR TITLE
enable editor changes in sketch mode, refactor some of the code manager

### PR DESCRIPTION
**Pull request by Irev-Dev**
…ges to t… (#2286)"

This reverts commit e7ab64526703a76e1d062d0c430fcf9363d8cfac.

implementing https://github.com/KittyCAD/modeling-app/pull/2283 again, but just making sure playwright passes before merging.

Will also revolve https://github.com/KittyCAD/modeling-app/issues/1558

https://github.com/KittyCAD/modeling-app/assets/29681384/ca3b32e7-9952-4450-8c90-9e4d0f3fbee7

### CODE REVIEW

**Comment by Irev-Dev** [comment]
my bet is [fefbd75](https://github.com/KittyCAD/modeling-app/pull/2287/commits/fefbd754ddd194b92e2756a46f069d946d5d0b98) will break a bunch of the onboarding and a few other things, but I think that's just because `updateCodeStateEditor` needs to be replaced with `updateCodeEditor` in most places, wanted to see where playwright is going to break though.

File: src/routes/Onboarding/FutureWork.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default function FutureWork() {
 
   useEffect(() => {
     // We do want to update both the state and editor here.
-    codeManager.updateCodeStateEditor(bracket)
```
Comment (jessfraz) [inline src/routes/Onboarding/FutureWork.tsx:15]: wait so im confused you dont want to update the state for these? if that state is now never used we can nukke it

Comment (Irev-Dev) [reply src/routes/Onboarding/FutureWork.tsx:15]: well this isn't my code so not super sure, but there were onboarding tests that failed before this change that are now passing so I think this is what we want.

Comment (Irev-Dev) [reply src/routes/Onboarding/FutureWork.tsx:15]: updateCodeEditor is still used in a couple of more places so I'll keep it for now.

**Pull request merged**